KeenClient/KeenClient.m

```diff
@@ -382,56 +344,47 @@ + (KeenClient*)sharedClient {
 # pragma mark - Geo stuff
 
 - (void)refreshCurrentLocation {
-    // only do this if geo is enabled
-    if (geoLocationEnabled == YES) {
-        KCLogInfo(@"Geo Location is enabled.");
-        // set up the location manager
-        if (self.locationManager == nil && [CLLocationManager locationServicesEnabled]) {
-            self.locationManager = [[CLLocationManager alloc] init];
-            self.locationManager.delegate = self;
-        }
+    if (geoLocationEnabled == NO) {
+        KCLogInfo(@"Geo Location is disabled.");
+        return;
+    }
 
-        // check for iOS 8 and provide appropriate authorization for location services
+    KCLogInfo(@"Geo Location is enabled.");
+    // set up the location manager
+    if (self.locationManager == nil && [CLLocationManager locationServicesEnabled]) {
+        self.locationManager = [CLLocationManager new];
+        self.locationManager.delegate = self;
+    }
 
-        if(self.locationManager != nil)
-        {
-            // If location services are already authorized, then just start monitoring.
-            CLAuthorizationStatus clAuthStatus = [CLLocationManager authorizationStatus];
-            if ([KeenClient isLocationAuthorized:clAuthStatus]) {
-                [self startMonitoringLocation];
-            }
-#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 80000
```

Comment:
I'm all for going iOS 8+. We could make a pass and update targets at some point to actually require this deployment target. The podspec is at iOS 6 right now and there's a spattering of different versions throughout the project. Technically this is a breaking change, so it would be a major revision and should probably go in a `vnext_major` branch.